src/Magnum/Math/Functions.h

```diff
@@ -410,6 +411,33 @@ template<std::size_t size, class T> inline Vector<size, T> ceil(const Vector<siz
     return out;
 }
 
+
+/**
+@brief Computes a Binomial Coefficient
+
+Given n elements computes the number of ways to choose k elements
+without replacement. The number of ways to do this is
+given by
+@f[
+    \begin{pmatrix} n \\ k \end{pmatrix} = \frac{n! \cdot (n-k)!}{k!} = \frac{n \cdot (n-1) \cdots (n - k + 1)}{k \cdots 1}.
+@f]
+*/
+template<class T> inline typename std::enable_if<IsIntegral<T>::value, T>::type binomialCoefficient(T n, T k) {
+    CORRADE_ASSERT(k >= 0, "k must be positive to compute the Binomial Coefficient", T{0});
+    if (k > n) return 0;
+    if (k * 2 > n)
+        k = n-k;
+    if (k == 0) return 1;
+
+    T result = n;
+    for(T i = 2; i <= k; ++i ) {
+        CORRADE_ASSERT(result < std::numeric_limits<T>::max()/(n-i+1), "Overflow In Binomial Coefficient", T{0});
```

Comment:
I'm always prefixing asserts with the API name to make it clearer where it happened even without a backtrace, and it could probably make sense to print the inputs also:

```cpp
"Math::binomialCoefficient(): overflow for (" << Debug::nospace << n << k << Debug::nospace << ")"
```